From: Sam Thursfield <sam.thursfield@codethink.co.uk>
Date: Fri, 28 Oct 2022 15:55:57 +0200
Subject: Add support for Python 3.11

Following the steps documented here:
https://github.com/apache/buildstream/pull/1738/

(cherry picked from commit 071fac27585c9252971ba7b10264b3fb6dd52e8d)
---
 tox.ini | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/tox.ini b/tox.ini
index a5d14a2..5e6cec4 100644
--- a/tox.ini
+++ b/tox.ini
@@ -2,7 +2,7 @@
 # Tox global configuration
 #
 [tox]
-envlist = py36-nocover,py37-nocover,py38-nocover,py39-nocover,py310-nocover
+envlist = py37,py{38,39,310,311}-nocover
 skip_missing_interpreters = true
 
 #
@@ -13,16 +13,16 @@ skip_missing_interpreters = true
 [testenv]
 commands =
     # Running with coverage reporting enabled
-    py{36,37,38,39,310}-!nocover: pytest --basetemp {envtmpdir} --cov=buildstream --cov-config .coveragerc {posargs}
-    py{36,37,38,39,310}-!nocover: mkdir -p .coverage-reports
-    py{36,37,38,39,310}-!nocover: mv {envtmpdir}/.coverage {toxinidir}/.coverage-reports/.coverage.{env:COVERAGE_PREFIX:}{envname}
+    py{36,37,38,39,310,311}-!nocover: pytest --basetemp {envtmpdir} --cov=buildstream --cov-config .coveragerc {posargs}
+    py{36,37,38,39,310,311}-!nocover: mkdir -p .coverage-reports
+    py{36,37,38,39,310,311}-!nocover: mv {envtmpdir}/.coverage {toxinidir}/.coverage-reports/.coverage.{env:COVERAGE_PREFIX:}{envname}
 
     # Running with coverage reporting disabled
-    py{36,37,38,39,310}-nocover: pytest --basetemp {envtmpdir} {posargs}
+    py{36,37,38,39,310,311}-nocover: pytest --basetemp {envtmpdir} {posargs}
 deps =
-    py{36,37,38,39,310}: -rrequirements/requirements.txt
-    py{36,37,38,39,310}: -rrequirements/dev-requirements.txt
-    py{36,37,38,39,310}: -rrequirements/plugin-requirements.txt
+    py{36,37,38,39,310,311}: -rrequirements/requirements.txt
+    py{36,37,38,39,310,311}: -rrequirements/dev-requirements.txt
+    py{36,37,38,39,310,311}: -rrequirements/plugin-requirements.txt
 
     # Only require coverage and pytest-cov when using it
     !nocover: -rrequirements/cov-requirements.txt
@@ -35,9 +35,9 @@ passenv =
 # These keys are not inherited by any other sections
 #
 setenv =
-    py{36,37,38,39,310}: COVERAGE_FILE = {envtmpdir}/.coverage
+    py{36,37,38,39,310,311}: COVERAGE_FILE = {envtmpdir}/.coverage
 whitelist_externals =
-    py{36,37,38,39,310}:
+    py{36,37,38,39,310,311}:
         mv
         mkdir
 
